File: c++/14713.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

vector<char> stack;
vector<char> ans;

int main() {
  freopen("example.txt", "r", stdin);

  string s;

  getline(cin, s);
  bool isWord = true;

  for (int i = 0; i < s.length(); i++) {
    if (isWord) {
      if (s[i] == '<') {
        isWord = false;
        while (!stack.empty()) {
          int a = stack.back();
          ans.push_back(a);
          stack.pop_back();
        }
        ans.push_back(s[i]);
      } else if (s[i] == ' ') {
        while (!stack.empty()) {
          int a = stack.back();
          ans.push_back(a);
          stack.pop_back();
        }
        ans.push_back(s[i]);
      } else {
        stack.push_back(s[i]);
      }
    } else {
      if (s[i] == '>') {
        isWord = true;
        ans.push_back(s[i]);
      } else {
        ans.push_back(s[i]);
      }
    }
  }
  while (!stack.empty()) {
    int a = stack.back();
    ans.push_back(a);
    stack.pop_back();
  }

  for (int i = 0; i < ans.size(); i++) {
    cout << ans[i];
  }

  return 0;
}

File: js/14713.js
const fs = require('fs');

const s = fs.readFileSync('example.txt').toString().trim();

const stack = [];
const answer = [];
let isWord = true;

for (let i = 0; i < s.length; i++) {
  if (isWord) {
    if (s[i] === '<') {
      isWord = false;
      while (stack.length !== 0) {
        answer.push(stack.pop());
      }
      answer.push(s[i]);
    } else if (s[i] === ' ') {
      while (stack.length !== 0) {
        answer.push(stack.pop());
      }
      answer.push(s[i]);
    } else {
      stack.push(s[i]);
    }
  } else {
    if (s[i] === '>') {
      isWord = true;
      answer.push(s[i]);
    } else {
      answer.push(s[i]);
    }
  }
}

while (stack.length !== 0) {
  answer.push(stack.pop());
}

console.log(answer.join(''));
